splice_machine/src/main/java/com/splicemachine/derby/vti/ORCRecordIterator.java

```
@@ -0,0 +1,246 @@
+			for (int i = 1; i <= execRow.nColumns(); i++) {
+				Object field = value_lst.get(i - 1);
+				value = null;
+				if (field != null)
+					value = field.toString();
+				// Note: Date is in same format as Splice Default,
+				// For timestamp format yyyy-mm-dd hh:mm:ss[.f...]., Check if
+				// Timestamp.valueOf(value) is required
+				returnRow.getColumn(i).setValue(value);
+			}
+			return new LocatedRow(returnRow);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/*
+	 * This checks if there are more files to process in the folder
+	 */
+	public boolean hasNextFile() {
+		try {
+			// If it is folder, check the file list iterator for more files
+			if (isDir)
+				return fileList.hasNext();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+		// If it is single file, return false
+		return false;
+	}
+
+	/*
+	 * Get the the next ORC fiel to process in the folder
+	 */
+	public Path nextFile() {
+		try {
+			// IF it is folder, get the next file from the file list iterator
+			if (isDir)
+				return fileList.next().getPath();
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+		return null;
+	}
+
+	/*
+	 * Returns Records in the next file
+	 */
+	public void getNextFileRecords() throws IOException {
+		// Close the current records iterator to release the resources for
+		// current file
+		records.close();
+
+		// Get the Record Reader for the next file
+		Reader reader = getReader(this.nextFile());
+		// set the records iterator for the new file
+		this.records = reader.rows();
+	}
+
+	/*
+	 * Gets the RecordReader of the specified file
+	 */
+	private Reader getReader(Path toProcessPath) throws IOException {
+		Reader reader = null;
+		if (toProcessPath == null) {
+			throw new IOException("Null File Path");
+		}
+		reader = OrcFile.createReader(filesystem, toProcessPath);
+
+		return reader;
+
+	}
+
+	@Override
+	public void remove() {
+		throw new RuntimeException("Not Implemented");
```

Please throw new UnsupportedOperationException() instead, since we won't be supporting removal.